src/object/rpc_csum.c

```diff
@@ -213,9 +211,17 @@ crt_proc_struct_dcs_iod_csums_adv(crt_proc_t proc, crt_proc_op_t proc_op,
 			if (unlikely(rc))
 				break;
 		}
+		if (rc != 0)
+			return rc;
 		D_FREE(iod_csum->ic_data);
 	}
 
+	rc = proc_struct_dcs_csum_info(proc, &iod_csum->ic_akey);
+	if (unlikely(rc)) {
+		D_FREE(iod_csum->ic_data);
```

Comment:
minor, D_FREE(iod_csum->ic_data) possibly call twice in L216 and L221, D_FREE will not check NULL internally, so better to check if it is NULL here.

Reply:
I disagree.  Both free() and D_FREE() accept NULL, and in fact we encourage calling D_FREE with null rather than checking to improve readability of both the code and the logs.  D_FREE() will set the pointer to NULL so as long as the same variable is called twice it's OK because the first one will have free it but the second one won't.